### PR TITLE
Per turn operation

### LINKED ISSUE #3: Strength Calculator
We require a system to determine the strength of a hand
The strength of the hand depends on the hand power and the symbol. 
Powers in order: 

- a flush (5-of-a-kind), worth a 12x multiplier
- four-of-a-kind, worth a 8x multiplier
- full house (3-of-a-kind plus a pair of another suit), worth a 6x multiplier
-  three-of-a-kind, worth a 4x multiplier
- two pair, worth a 3x multiplier
- one pair, worth a 2x multiplier
- high card, worth a 1x multiplier

In the case a player and Luigi have hands of the same strength, the suits will act as a tiebreaker. 

This should probably be implemented in Java.

### LINKED ISSUE #4: Turn Order
We need a system to determine turn order. 
Turn order will be determined by tokens possessed, then hands won, then finally the player number. 

Once everything is implemented, turn order should be relatively straightforward.

### LINKED ISSUE #5: Individual Turn
A system to carry out a turn is needed. 
This involves raising, card changing, etc.  

Specifically, during a turn, players will each take turns betting up to 5 tokens, then selecting zero to five cards from their hand to discard and redraw. 

This probably can all be done in Java

### PR DESCRIPTION
**Pull request by MadTurt1e**
Game has been updated  - this is the semi-functioning version of the game. 
Note arraylists have not been implemented in this - I find using them a little too "quirky". This can of course be changed in a potential future version. 
In any case, the core game elements have been implemented in this - 
#3 and #5 come to mind, as well as #17 (though that is not yet fully implemented). 

Once this PR is accepted, 
Closes #3, Closes #5, Closes #4

### CODE REVIEW

File: src/main/java/game/jdbc/picturepoker/Game.java
Comment (jeffreyw2025) [inline src/main/java/game/jdbc/picturepoker/Game.java:1]: Little miffed about the ArrayList thing, but should be fine as long as we have a method to sort. Take advantage of my comparison method!

File: src/main/java/game/jdbc/picturepoker/GamePlay.java
Comment (jeffreyw2025) [inline src/main/java/game/jdbc/picturepoker/GamePlay.java:1]: I created a raise() function so there is a systematic way to handle betting, but I guess it's OK to just use it with a UI later on.

Bet logic is slightly bugged- it allows players to bet more coins than they have if they have fewer than 5 coins. Please fix this.

I created a dealer_cards table that should interface with the Game class. Are you sure it doesn't work?

The payout logic double-penalizes players if they fail. The method should return 0 if they lose to Luigi instead. However, I will say that the hand evaluation method is pretty clever, but do double-check some potential edge-cases just to make sure it works.

**Review by jeffreyw2025 — CHANGES_REQUESTED** [review]
Most of the functionality seems fine, just some of the betting and payout logic needs refinement. See my comment on GamePlay.java.

File: src/main/java/game/jdbc/picturepoker/Game.java
Comment (MadTurt1e) [reply src/main/java/game/jdbc/picturepoker/Game.java:1]: absolutely right - I'll try using the compare function to do player order (i didn't notice this before)

**Comment by MadTurt1e** [comment]
I've added player turns - this closes #4
Of course, we need to add some more fine-grained turn orders, but we really don't as the implementation adapts for this already.

File: src/main/java/game/jdbc/picturepoker/GamePlay.java
Comment (MadTurt1e) [reply src/main/java/game/jdbc/picturepoker/GamePlay.java:1]: Most functions mentioned have been implemented.

File: src/main/java/game/jdbc/picturepoker/Game.java
Comment (MadTurt1e) [reply src/main/java/game/jdbc/picturepoker/Game.java:1]: Player order added

**Review by jeffreyw2025 — CHANGES_REQUESTED** [review]
Ok, this is super nitpicky, but the hand evaluation fails in a few scenarios. Consider the Player hand {Mushroom, Mushroom, Mushroom, Cloud, Cloud} vs the Luigi hand {Cloud, Cloud, Cloud, Star, Star}. By your logic, Player gets 4030 points (4000 for FH + 30 for 3 mushrooms) vs Luigi's 4100 (4000 for FH + 100 for stars), so the player would lose. However, the actual tiebreaker evaluates the three of a kinds first, so the Player would win. I know it's asking a lot but could you make sure the hand evaluation matches with the actual tiebreakers from the DS games?

**Comment by MadTurt1e** [comment]
okay, done.

**Comment by jeffreyw2025** [comment]
I'm so sorry, Richard, but my grader instincts have given me another counterexample. Consider the Player hand {Star, Star, Cloud, Cloud, Mario} versus Luigi hand {Mario, Mario, Luigi, Luigi, Star}. The player strength evaluates to two pair + 40(6+1) vs Luigi's strength of two pair + 40(5+4), with the player losing again. However, the tiebreaker is to compare the stronger pairs first, so the player should win. I know I'm being really mean but I want this to be perfect. Make sure all the edge cases are fixed this time!

**Comment by MadTurt1e** [comment]
Would it be possible for you to come up with an algorithm for point calculation?
 I have been working on the (relatively more pressing) update and delete functions and really cannot think of any reasonable ways to do point calculations and account for every edge case as of now, so a new perspective could be helpful in this scenario.
I appreciate it!

**Pull request closed without merging**